qdrant_client/local/local_collection.py

```diff
@@ -2163,10 +2170,20 @@ def _upsert_point(self, point: models.PointStruct) -> None:
 
     def upsert(self, points: Union[Sequence[models.PointStruct], models.Batch]) -> None:
         if isinstance(points, list):
+            if len(points) > self.LARGE_DATA_THRESHOLD:
+                deprecation_warning_once(
+                    f"Local mode is not recommended for collections with more than {self.LARGE_DATA_THRESHOLD:,} points. Consider using Qdrant server mode for better performance with large datasets.",
+                    idx="large-data-local",
+                )
             for point in points:
                 self._upsert_point(point)
         elif isinstance(points, models.Batch):
             batch = points
+            if len(batch.ids) > self.LARGE_DATA_THRESHOLD:
+                deprecation_warning_once(
+                    f"Local mode is not recommended for collections with more than {self.LARGE_DATA_THRESHOLD:,} points. Consider using Qdrant server mode for better performance with large datasets.",
+                    idx="large-data-local",
+                )
```

Comment:
it's not about the amount of points they're trying to upsert with one call, it's about the amount of points in the collection in general